geth/common/types.go

```diff
@@ -173,28 +173,11 @@ type TxQueueManager interface {
 	DiscardTransactions(ids string) map[string]RawDiscardTransactionResult
 }
 
-// JailExecutor defines an interface which exposes method to be executed
-// against a Jail vm.
-type JailExecutor interface {
-	// Run exist so we are able to execute js code on pure otto.VM without runing
-	// it on the event loop.
-	Run(string) (otto.Value, error)
-
-	// Exec exists for the purpose to execute has normal on the event loop provided by
-	// ottoext.
-	Exec(string) (otto.Value, error)
-
-	// Fetch calls the underlying FetchAPI which makes http request
-	// to desired path. (See https://developer.mozilla.org/en/docs/Web/API/Fetch_API).
-	Fetch(string, func(otto.Value)) (otto.Value, error)
-}
-
 // JailCell represents single jail cell, which is basically a JavaScript VM.
 type JailCell interface {
-	CellVM() *otto.Otto
-	CellLoop() *loop.Loop
-	Executor() JailExecutor
-	Copy() (JailCell, error)
+	VM() *otto.Otto
```

Comment:
Do we really need this interface? If we remove it, we lose nothing because it's only returned in tests and used nowhere as an argument. We'll also be able to remove `Jail.GetCell` and rename `Jail.getCell` into `Jail.GetCell`.

As I can see, its only purpose is to encapsulate `vm` and `loop` fields and:
1. It is not a recommended practice in Go.
2. It adds unnecessary complexity.
3. `VM` and `Loop` methods are only called in unit tests which can easily be replaced with "real" fields.
4. It will allow you to get rid of unneeded type assertions.